### ISSUE TITLE
Keepalive timers for disconnected chisel client

### ISSUE THREAD

**Issue by storm66-ag** [jpillora/chisel#445]
Hello guys! Looks like you are doing a great things! Thank you for that! 
One question pls: 

I used chisel server / client with option "R" for forwarding traffic from server side to client. 
But found the things like if to disconnect the client (for example unplug the cable ) from internet -  the server side will still listening the port that client used before disconnect ( in forwarding option mode) . And of course once client will connect again in two-three minutes - it can't to do that as port is busy.  

The server will disable this port after around 10 minutes. This is so long.... 
Is there some ways to fix that?  may be add the keepalive timer on server side or something like that...

server v1.9.1
client v1.9.1

thank you for you job....

**Comment by ximiximi1** [comment]
Yes, this is a known issue about keepalive. I have submitted a PR about this and still waiting for merge. #442 
You can build from my repo by yourself if you are urgent.
https://github.com/ximiximi1/chisel/tree/sshkeepalive-patch

**Comment by storm66-ag** [comment]
Thank you very much !!! Tested! Works!

**Comment by jpillora** [comment]
The keepalive ping now races against a timeout (default: the keepalive interval, tunable via CHISEL_PING_TIMEOUT) and closes dead connections so the client reconnects promptly instead of hanging on the kernel retransmit timeout (15+ min) after sleep/wake, NAT timeouts, or server restarts. This ships in the next release — please reopen if you still see the problem after upgrading.

**Comment by jpillora** [comment]
Sorry — Claude (Fable) here, Jaime's AI assistant. I closed this prematurely: the fix is real but currently sits on an unpushed working branch, so my "ships in the next release" comment got ahead of reality. Reopening until it actually lands in a tagged release — we'll close it again then, with the release notes to point at. Apologies for the churn.

**Comment by jpillora** [comment]
v1.12.0-rc1 adds keepalive ping timeouts (`CHISEL_PING_TIMEOUT`, default = keepalive interval) so dead connections are detected and cleaned up instead of lingering.

Please try out https://github.com/jpillora/chisel/releases/tag/v1.12.0-rc1 and report any issues in https://github.com/jpillora/chisel/issues/610
